jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/Cloudformation.kt

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudformation
+
+import software.amazon.awssdk.services.cloudformation.CloudFormationClient
+import software.amazon.awssdk.services.cloudformation.model.CloudFormationException
+import software.amazon.awssdk.services.cloudformation.model.DescribeStacksRequest
+import software.amazon.awssdk.services.cloudformation.model.StackStatus
+import software.aws.toolkits.resources.message
+import java.util.concurrent.TimeoutException
+
+fun CloudFormationClient.waitForStackDeletionComplete(stackName: String, maxAttempts: Int, delay: Int) {
+    val deletionSuccessStatuses = setOf(StackStatus.DELETE_COMPLETE)
+    val deletionFailedStatuses = setOf(StackStatus.DELETE_FAILED, StackStatus.ROLLBACK_FAILED, StackStatus.CREATE_FAILED, StackStatus.UPDATE_ROLLBACK_IN_PROGRESS, StackStatus.UPDATE_ROLLBACK_FAILED)
+
+    repeat(maxAttempts) {
+        val stacks = try {
+            this.describeStacks(DescribeStacksRequest.builder().stackName(stackName).build()).stacks()
+        } catch (e: CloudFormationException) {
+            if (e.awsErrorDetails().errorCode() == "ValidationError") {
+                return
+            }
+            throw e
+        }
+        if (stacks.size != 1) {
+            throw RuntimeException(message("cloudformation.delete_stack.multiple_stacks_with_same_name"))
+        }
+
+        val targetStack = stacks[0]
+        when (targetStack.stackStatus()) {
+            in deletionSuccessStatuses -> return
+            in deletionFailedStatuses -> throw RuntimeException(message("cloudformation.delete_stack.failed", stackName, targetStack.stackStatus().toString()))
+            else -> Thread.sleep(delay * 1000L)
+        }
+    }
+    throw TimeoutException(message("cloudformation.delete_stack.timeout", stackName, delay * maxAttempts))
+}
+
+// TODO Use Java SDK waiter once supported
+fun CloudFormationClient.waitForStackDeletionComplete(stackName: String) {
```

Comment:
Why do we need this? Why can't we just use default parameters on the method above? eg.

```
CloudFormationClient.waitForStackDeletionComplete(stackName: String, maxAttempts: Int = 120, delay: Duration = Duration.ofSeconds(20))
```

Reply:
It's for the test. In the test case, I don't want to wait for so long for time out.

Reply:
what's your point? we can still collapse them into one method using Kotlin default arguments.

Reply:
I see your point. Done.